packages/DateRangePicker/src/DateRangePicker.js

```diff
@@ -0,0 +1,238 @@
+  onDatesChange: PropTypes.func.isRequired,
+
+  children: PropTypes.node,
+};
+
+const defaultProps = {
+  startDate: null,
+  endDate: null,
+  children: null,
+};
+
+const DateRangePicker = ({ startDate, endDate, onDatesChange, children }) => {
+  const [shouldShowPopover, setShouldShowPopover] = React.useState(false);
+  const [currentFocus, setCurrentFocus] = React.useState(null);
+  const [possibleStartDate, setPossibleStartDate] = React.useState(startDate || moment());
+  const [possibleEndDate, setPossibleEndDate] = React.useState(endDate || moment().add(1, "months"));
+
+  const popoverContentRef = React.useRef(null);
+  const startDateInputRef = React.useRef(null);
+  const endDateInputRef = React.useRef(null);
+
+  const I18n = useI18n();
+  const debouncedPossibleStartDate = useDebounce(possibleStartDate, 300);
+  const debouncedPossibleEndDate = useDebounce(possibleEndDate, 300);
+
+  React.useEffect(() => {
+    setPossibleStartDate(startDate || moment());
+  }, [startDate]);
+
+  React.useEffect(() => {
+    setPossibleEndDate(endDate || moment().add(1, "months"));
+  }, [endDate]);
+
+  const {
+    "DateRangePicker.Input": { props: inputProps = {} } = {},
+    "DateRangePicker.StartInput": {
+      props: { placeholder: startInputPlaceholder = I18n.t("daterange.start_date"), ...startInputProps } = {},
+    } = {},
+    "DateRangePicker.EndInput": {
+      props: { placeholder: endInputPlaceholder = I18n.t("daterange.end_date"), ...endInputProps } = {},
+    } = {},
+    "DateRangePicker.Popover": { props: popoverProps = {} } = {},
+  } = extractChildren(children, [
+    "DateRangePicker.Input",
+    "DateRangePicker.StartInput",
+    "DateRangePicker.EndInput",
+    "DateRangePicker.Popover",
+  ]);
+
+  function isAnyElementContainsFocus() {
+    return (
+      isElementContainsFocus(popoverContentRef.current) ||
+      isElementContainsFocus(startDateInputRef.current) ||
+      isElementContainsFocus(endDateInputRef.current)
+    );
+  }
+
+  function showPopover() {
+    if (!shouldShowPopover) setShouldShowPopover(true);
+  }
+
+  function hidePopover() {
+    if (shouldShowPopover) setShouldShowPopover(false);
+    if (startDateInputRef.current) startDateInputRef.current.blur();
+    if (endDateInputRef.current) endDateInputRef.current.blur();
+    setCurrentFocus(null);
+  }
+
+  function handleClosePopover() {
+    if (!isAnyElementContainsFocus()) {
+      hidePopover();
+    }
+  }
+
+  function handleKeyUpOnEscape(event) {
+    if (event.key === "Escape") {
+      hidePopover();
+    }
+  }
+
+  function onDateInputFocus(type) {
+    return () => {
+      setCurrentFocus(type);
+    };
+  }
+
+  function onFocusChange(type) {
+    if (type === START_DATE) {
+      startDateInputRef.current.focus();
+    } else if (type === END_DATE) {
+      endDateInputRef.current.focus();
+    } else {
+      hidePopover();
+    }
+  }
+
+  function onChangeStartDate(newDate) {
+    onDatesChange({ startDate: newDate, endDate });
```

Comment:
yeah I think now that I see the response from this callback, would be beneficial to named `onChange` as keep consistency on the library:
```ts
 onChange({ startDate, endDate } : { starDate: moment.Moment, endDate: moment.Moment }): void
```